<div style="border-left: 8px solid #1A365D; border-right: 2px solid #E2E8F0; border-top: 2px solid #E2E8F0; border-bottom: 2px solid #E2E8F0; padding: 25px; border-radius: 4px; background-color: #F8FAFC; font-family: 'Segoe UI', Arial, sans-serif; box-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.05);">
    <span style="color: #D69E2E; font-weight: bold; letter-spacing: 1.5px; font-size: 0.9em; text-transform: uppercase;">FES Acatlán • UNAM</span>
    <h1 style="color: #1A365D; font-size: 2.2em; font-weight: 700; margin-top: 5px; margin-bottom: 5px; line-height: 1.2;">Diplomado en Técnicas Estadísticas y Minería de Datos</h1>
    <h3 style="color: #4A5568; font-size: 1.3em; font-weight: 400; margin-top: 0; margin-bottom: 20px;">Módulo II: Modelos Estadísticos</h3>
    <hr style="border: 0; height: 1px; background: #CBD5E0; margin-bottom: 20px;">
    <table style="width: 100%; border-collapse: collapse; font-size: 0.95em; color: #4A5568;">
        <tr>
            <td style="padding: 4px 0; width: 50%;"><strong>Notebook:</strong> Estadística Indiferencial</td>
            <td style="padding: 4px 0; width: 50%; text-align: right;"><strong>Estudiante:</strong> Blanca Sánchez Germán</td>
        </tr>
        <tr>
            <td style="padding: 4px 0;"><strong>Profesora:</strong> Act. Claudia C. Reyes Montes de Oca</td>
            <td style="padding: 4px 0; text-align: right;"><strong>Fecha:</strong> 4 al 11 de Julio 2026</td>
        </tr>
    </table>
</div>

# **<span style="color:#191970;"> Inferencia Estadística y Métodos de Estimación</span>**

El propósito de la estadística es utilizar la información de una muestra aleatoria para realizar inferencias sobre una población con parámetros desconocidos. El parámetro de interés en el experimento se denomina **parámetro objetivo**.

* **<span style="color:#191970;">Parámetros Poblacionales:</span>** Valores fijos y desconocidos
  * $\mu$ para la media
  * $\sigma^2$ para la varianza.
* **<span style="color:#191970;">Muestra Aleatoria:</span>** Conjunto de variables aleatorias $(X_1, \dots, X_n)$ obtenidas de una densidad conocida
  $$f(\cdot; \theta)$$,
donde $\theta$ es el parámetro desconocido.
* **<span style="color:#191970;">Estadísticos Muestrales:</span>** Variables aleatorias que son funciones de la muestra.
  *  Media muestral.
    $$\bar{X} = \frac{\sum X_i}{n}$$ 
  *  Varianza muestral.
    $$S^2 = \frac{\sum (X_i - \bar{X})^2}{n-1}$$


---

# **<span style="color:#191970;"> Teoría de Estimadores</span>**

Un **<span style="color:#191970;">estimador</span>** es una función de la muestra aleatoria utilizada para aproximar un parámetro poblacional desconocido $\theta$

Sea una muestra aleatoria:

$$X_1,X_2,\dots,X_n$$

obtenida de una población con distribución:

$$f(x;\theta)$$

donde $\theta$ representa un parámetro poblacional desconocido. Un **<span style="color:#191970;">estimador</span>** de $\theta$ es una función de las observaciones muestrales:

$$\hat{\theta}=T(X_1,X_2,\dots,X_n)$$

donde $T$ es un **estadístico muestral** que no depende de parámetros desconocidos.

El estimador es una **variable aleatoria**, ya que depende de los valores que toma la muestra aleatoria. Su finalidad es proporcionar una aproximación al valor verdadero del parámetro poblacional que se desea conocer.

Por ejemplo:

* Para estimar la media poblacional $\mu$, se utiliza la media muestral:

$$\hat{\mu}=\bar{X}=\frac{1}{n}\sum_{i=1}^{n}X_i$$

* Para estimar la varianza poblacional $\sigma^2$, se utiliza la varianza muestral:

$$\hat{\sigma}^{2}=S^2=\frac{1}{n-1}\sum_{i=1}^{n}(X_i-\bar{X})^2$$


## <span style="color:#191970;">Distinción Fundamental</span>
* **Estimador:** Se refiere a la **función** o regla matemática utilizada, la cual es una variable aleatoria.
* **Estimación:** Es el **valor específico** que toma dicha función al sustituir los datos observados.
  
#### <span style="color:#191970;">Ejemplo</span>

Supongamos que se desea estimar la edad promedio $\mu$ de una población.

Se obtiene una muestra de cinco edades:

$$18,20,22,25,30$$

El **estimador** utilizado es la media muestral:

$$\bar{X}=\frac{\sum X_i}{n}$$

Al sustituir los valores observados se obtiene la **estimación**:

$$\bar{x}=\frac{18+20+22+25+30}{5}=23$$

Por lo tanto:

* $\bar{X}$ → **Estimador** (función aleatoria antes de observar los datos).
* $\bar{x}=23$ → **Estimación** (valor calculado a partir de la muestra).



---
## **<span style="color:#191970;"> Propiedaddes de los Estimadores </span>**

Para evaluar si un estimador es "bueno" o "malo", utilizamos estas propiedades fundamentales que garantizan la confiabilidad de nuestras inferencias:

| Propiedad | Definición Formal | Interpretación | Ejemplo de Aplicación |
| :--- | :--- | :--- | :--- |
| **Insesgadez** | Un estimador es insesgado si su valor esperado coincide con el parámetro poblacional ($\mathbb{E}[\hat{\Theta}] = \theta$)| No presenta errores sistemáticos; en promedio, coincide con el valor real. | La media muestral ($\bar{X}$) para estimar la media poblacional ($\mu$). |
| **Consistencia** |Un estimador es consistente si converge en probabilidad al valor real del parámetro a medida que el tamaño de muestra ($n$) tiende a infinito $\hat{\Theta}_n \xrightarrow{p} \theta$ | A mayor cantidad de datos, el estimador converge al valor verdadero. | Usar una muestra grande para estimar la proporción de votos en una elección. |
| **Eficiencia** | Entre varios estimadores insesgados, el más eficiente es aquel que posee la menor varianza ($\text{Var}(\hat{\Theta})$) | Es el más estable y preciso ante variaciones muestrales. | Comparar dos estimadores insesgados y elegir el de menor varianza. |
| **Suficiencia** | $f(x|\hat{\Theta})$ independiente de $\theta$ | Contiene toda la información de la muestra sobre el parámetro. | La muestra completa es suficiente para estimar la media de una distribución normal. |
| **Completitud** | $\mathbb{E}[g(\hat{\Theta})] = 0 \implies g(\hat{\Theta}) = 0$ | Asegura la unicidad del estimador de mínima varianza (UMVUE). | Garantiza que no hay otro estimador mejor que el que ya encontramos. |

---

# **<span style="color:#191970;"> Método de los Momentos (MM)</span>**

Consiste en igualar los momentos muestrales con los correspondientes momentos poblacionales y resolver esta ecuación, o sistema de ecuaciones, para el parámetro o vector de parámetros, cuando ello sea posible. 

El método fue introducido por **Karl Pearson** a principios del siglo XX. Es un mecanismo general para encontrar estimadores de un parámetro o vector de parámetros $\theta$, de los cuales depende una función de densidad o probabilidad $f(x, \theta)$.

### <span style="color:#191970;">Fundamentos y Justificación</span>
* **Igualación:** El método iguala momentos poblacionales ($\mathbb{E}[X^k]$) con momentos muestrales ($m_k = \frac{1}{n} \sum_{i=1}^{n} X_i^k$).
* **Ley de los Grandes Números:** Cuando $n$ es grande, el momento muestral es cercano al poblacional.
* **Aplicación:** Funciona tanto para distribuciones discretas como continuas.

### <span style="color:#191970;">Consideraciones y Limitaciones</span>
*  Requiere que los momentos existan y tengan expresiones sencillas. Si no existen momentos finitos, el método no puede aplicarse.
*  Asume que existe una única solución sencilla, lo cual no siempre ocurre con sistemas no lineales complejos.

### <span style="color:#191970;">Ejemplos e Interpretación</span>
* **Interpretación:** Es un método intuitivo que usa características básicas (como la media y varianza) para inferir parámetros poblacionales a partir de los datos observados.
* **Ejemplo (Gamma $\alpha, \beta$):** Resulta en $\hat{\alpha} = \bar{X}^2 / S_n^2$ y $\hat{\beta} = \bar{X} / S_n^2$.

---

### <span style="color:#191970;">Tabla de Referencia: Estimadores por el Método de Momentos</span>

| Distribución | Parámetro(s) | Estimador(es) |
| :--- | :--- | :--- |
| **Bernoulli** | $p \in (0, 1)$ | $\hat{p} = \bar{X}$ |
| **Poisson** | $\lambda \in (0, \infty)$ | $\hat{\lambda} = \bar{X}$ |
| **Exponencial** | $\lambda \in (0, \infty)$ | $\hat{\lambda} = 1 / \bar{X}$ |
| **Normal** | $\mu, \sigma^2$ | $\hat{\mu} = \bar{X}, \hat{\sigma}^2 = \frac{n-1}{n} S^2$ |
| **Chi-cuadrada** | $k \in (0, \infty)$ | $\hat{k} = \bar{X}$ |
| **Gamma** | $\gamma, \lambda$ | $\hat{\gamma} = \frac{m_1^2}{m_2 - m_1^2}, \hat{\lambda} = \frac{m_1}{m_2 - m_1^2}$ |
| **Uniforme (cont.)**| $a < b$ | $\hat{a} = \frac{4m_1^2 - 3m_2}{2m_1 - 1}, \hat{b} = \frac{3m_2 - 2m_1}{2m_1 - 1}$ |
| **Beta** | $a, b$ | $\hat{a} = \frac{m_1(m_1 - m_2)}{m_2 - m_1^2}, \hat{b} = \frac{(1 - m_1)(m_1 - m_2)}{m_2 - m_1^2}$ |

*(Nota: $m_k$ representa el $k$-ésimo momento muestral, donde $m_1 = \bar{X}$)*.

# **<span style="color:#191970;"> Método de Máxima Verosimilitud (MLE / MMV)</span>**

Consiste en encontrar el valor de $\theta$ que maximiza a la funci ́on $L(\theta)$. Al valor de θ en donde Lpθq alcanza su m ́aximo se le llama estimaci ́on de m ́axima verosimilitud o estimaci ́on m ́aximo veros ́ımil.

### <span style="color:#191970;"> Ejemplo de la Urna</span>
Considera una urna con pelotas blancas y negras en una proporción desconocida ($p=0.25$ o $p=0.75$). Si extraemos una muestra de $n=3$ pelotas, el estimador elige el valor de $p$ que maximiza la probabilidad del resultado obtenido.
*  Si al extraer 3 pelotas obtenemos $x=0$ pelotas negras, preferimos el parámetro $p=0.25$ sobre $p=0.75$, porque la probabilidad de obtener ese resultado es mayor con $p=0.25$.
*  El estimador selecciona para cada resultado $x$ el valor de $p$ tal que $f(x; \hat{p}) > f(x; p')$ para cualquier otro valor posible $p'$.

---

## <span style="color:#191970;">Fundamentos Formales</span>

Sea $(X_1, \dots, X_n)$ una muestra aleatoria con función de densidad o probabilidad $f(x; \theta)$. 

* **Función de Densidad Conjunta:** Corresponde al producto de las funciones individuales debido a la independencia de la muestra: $f(X_1, \dots, X_n; \theta)$.
* **Función de Verosimilitud ($L(\theta)$):** Se construye multiplicando todas las densidades marginales evaluadas en la muestra:
    $$L(\theta) = \prod_{i=1}^{n} f(x_i; \theta) = f(x_1; \theta) \cdot f(x_2; \theta) \cdots f(x_n; \theta)$$
* **Log-Verosimilitud ($\log(L(\theta))$):** Aplicamos el logaritmo para transformar el producto en una sumatoria algebraica:
    $$\log(L(\theta)) = \log\left(\prod_{i=1}^{n} f(x_i; \theta)\right) = \sum_{i=1}^{n} \log(f(x_i; \theta))$$

## <span style="color:#191970;">Optimización Matemática</span>
Para hallar el estimador ($\hat{\theta}_{mv}$), derivamos respecto a $\theta$ e igualamos a cero:
$$\frac{d\log(L(\theta))}{d\theta} = 0$$

# <span style="color:#191970;">Ejemplo 1: Bernoulli($p$)</span>
Sea $(X_1, \dots, X_n)$ una muestra aleatoria procedente de una distribución Bernoulli con función de masa de probabilidad: $f(x; p) = p^x(1-p)^{1-x}$.

**1. Construcción de la Verosimilitud:**
$$L(p) = \prod_{i=1}^{n} p^{x_i}(1-p)^{1-x_i} = p^{\sum X_i}(1-p)^{n - \sum X_i}$$

**2. Aplicación de Logaritmo:**
$$\ln L(p) = \ln\left(p^{\sum X_i}\right) + \ln\left((1-p)^{n - \sum X_i}\right) = \left(\sum X_i\right)\ln(p) + \left(n - \sum X_i\right)\ln(1-p)$$

**3. Derivación e Igualación a Cero:**
$$\frac{d\ln L(p)}{dp} = \frac{\sum X_i}{p} - \frac{n - \sum X_i}{1-p} = 0$$

Multiplicando cruzado para resolver la ecuación lineal y despejar $p$:
$$(1-p)\sum X_i - p\left(n - \sum X_i\right) = 0$$
$$\sum X_i - p\sum X_i - np + p\sum X_i = 0 \implies \sum X_i - np = 0$$
$$\hat{p} = \frac{\sum X_i}{n} = \bar{X}$$

**Conclusión** El estimador de máxima verosimilitud para la probabilidad de éxito $p$ en una distribución Bernoulli es la media muestral ($\bar{X}$).

In [1]:
import numpy as np
from scipy.stats import bernoulli

#  Generamos una muestra aleatoria de una Bernoulli con p real = 0.65
p_real = 0.65
np.random.seed(123)
muestra_bernoulli = bernoulli.rvs(p=p_real, size=500)

#  Aplicamos el estimador MLE obtenido analíticamente en tus apuntes (p_hat = x_barra)
p_estimado_mle = np.mean(muestra_bernoulli)

print("   ESTIMACIÓN POR MÁXIMA VEROSIMILITUD (BERNOULLI)   ")
print(f"Tamaño de muestra n = {len(muestra_bernoulli)}")
print(f"Parámetro poblacional real p = {p_real}")
print(f"Estimador obtenido por MLE (p_hat = media): {p_estimado_mle:.4f}")

   ESTIMACIÓN POR MÁXIMA VEROSIMILITUD (BERNOULLI)   
Tamaño de muestra n = 500
Parámetro poblacional real p = 0.65
Estimador obtenido por MLE (p_hat = media): 0.6600


## <span style="color:#191970;">Interpretación</span>
Se obtuvo una estimación de $0.6600$ para un parámetro real de $0.65$.
El estimador de máxima verosimilitud (la media muestral) funciona correctamente. La pequeña diferencia ($0.01$) se debe a la variabilidad natural del muestreo aleatorio. Esto demuestra que, al tener una muestra grande ($n=500$), el estimador es **consistente** y se aproxima significativamente al valor poblacional.


# <span style="color:#191970;">Ejemplo 2: Lanzamineto de una moneda </span>

Lanzamos una moneda 100 veces, sabiendo que se obtienen 55 soles. Encontrar el EMV para la probabilidad $p$ de obtener sol en un solo lanzamiento.

Sea
$X$: el número de soles que se obtienen al lanzar una moneda se tiene que $X\sim \text{Bin}(100,p)$

**Función de Densidad (Masa de Probabilidad):**
La función de probabilidad de $X$ para obtener exactamente $k$ éxitos (soles) es:
$$f(k) = P(X = 55) = \binom{100}{55} p^{55} (1-p)^{100-55}$$
para $k \in \{0, 1, 2, \dots, 100\}$.

**Esperanza (Valor Esperado):**
La media o valor esperado de una distribución binomial es $E[X] = n \cdot p$. Para este caso:
$$E[X] = 100p$$

**Varianza:**
La varianza mide la dispersión y se calcula como $Var(X) = n \cdot p \cdot (1-p)$. Para este caso:
$$Var(X) = 100p(1-p)$$


In [2]:
from scipy.stats import binom
import numpy as np

# vamos a utilizar la libreria sympy para matemática simbólica
import sympy

# importar la variable simbólica x
from sympy.abc import x

In [5]:
# Vamos a definir a p como una variable simbólica positiva: 0<p<1
p = sympy.symbols('p', positive=True)

# Definimos la función de verosimilitud
# se puede omitir la constante
f = p**55 * (1-p)**45

# ahora derivamos f con respecto a p 
df = sympy.diff(f,p)

# Hay que igualar la derivada a cero y despejar a p 
# df=0 para encontar p
phat = sympy.solve(df, p) [0]

# Mostramos los resultados
print("Estimador Máximo Verosímilitud (EMV) para p es:", phat)


Estimador Máximo Verosímilitud (EMV) para p es: 11/20


## <span style="color:#191970;">Interpretación</span>
El estimador máximo verosímil calculado simbólicamente es $11/20$.

Dado que obtuvimos 55 éxitos en 100 intentos, el resultado de $0.55$ es la frecuencia relativa observada. El método de máxima verosimilitud confirma matemáticamente que, para una distribución Binomial, la mejor estimación de la probabilidad de éxito es simplemente la proporción de éxitos en la muestra.

# <span style="color:#191970;">Ejemplo 3 </span>

Vamos a simular una muestra de tamaño 100 de ceros y unos.

Sea $X_1, X_2, \dots, X_{100}$ una muestra aleatoria de tamaño $n = 100$ tal que cada $X_i \sim \text{Bernoulli}(p)$, es decir, un ensayo de Bernoulli ($n=1$).

La función de verosimilitud para la muestra es:
$$L(p) = \prod_{i=1}^{100} f(x_i; p) = \prod_{i=1}^{100} p^{x_i} (1-p)^{1-x_i} = p^{\sum x_i} (1-p)^{100 - \sum x_i}$$

El Estimador Máximo Verosímil (EMV) óptimo para este caso corresponde a la media muestral:
$$\hat{p} = \bar{X} = \frac{\sum_{i=1}^{100} X_i}{100}$$

In [6]:
# n=1 : un ensayo de Bernoulli
# p : es la probabilidad de éxito
# size : genera 100 observaciones
r = binom.rvs(n=1, p=0.7, size = 100)
r

array([1, 1, 0, 1, 0, 1, 1, 1, 0, 1, 0, 0, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1,
       0, 1, 1, 1, 1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 0,
       1, 0, 1, 1, 1, 0, 1, 1, 1, 0, 1, 0, 1, 1, 1, 1, 1, 1, 0, 1, 0, 1,
       1, 1, 0, 1, 1, 1, 1, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 1, 1,
       0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1])

In [7]:
# Vamos a a caclular EMV

# Definimos la función de verosimilitud para una sola observación
f = p**x * (1 - p)**(1 - x)

#Calculamos el producto de todods los elementos 
# SUBS(x,i) sustituye  la variable simbolica x por un valor concreto de la muestra 
J = np.prod([f.subs(x,i) for i in r])

# Calculamos el log
log_J = sympy.expand_log(sympy.log(J))

# Calculamos la derivada respecto a p
phat = sympy.solve(sympy.diff(log_J,p), p)[0]

print("Función de verosimilitud de la muestra (J):",phat)


Función de verosimilitud de la muestra (J): 71/100


In [8]:
#podemos vereficar las soluciones con: 
sol =sympy.solve(sympy.diff(log_J,p), p)
sol

[71/100]

## <span style="color:#191970;">Interpretación</span>

*Resultado:** El código obtuvo $71/100$  a partir de una simulación generada con $p=0.7$.
Al construir la función de verosimilitud de forma explícita mediante el producto de las probabilidades individuales, el sistema llega al mismo resultado que la media muestral. Este ejercicio valida que el método numérico de optimización coincide con el método analítico, demostrando que la estimación es la más "verosímil" o probable de haber producido los datos observados en nuestra muestra de 100 elementos.

# <span style="color:#191970;">Ejemplo 4</span>

El tiempo de retardo para pacientes con enfermedad coronaria potencial se define como la duración entre el comienzo de un síntoma agudo cardiovascular y la llegada a la unidad de emergencia.

Supongamos que el tiempo de retardo tiene una distribución exponencial de parámetro $\theta$.

Se tienen los siguientes registros de retardo (en minutos) en la clínica para los primeros 20 pacientes:
$$525, 719, 2880, 150, 30, 251, 45, 858, 15, 47, 90, 56, 68, 6, 189, 180, 60, 60, 294, 747$$

* Encontrar el estimador por el método de momentos para la media
* Encontrar el valor del estimador con los datos.

Se tiene que 
$$X : \text{tiempo de retardo (en minutos)}$$
y que $X \sim \text{Exponencial}(\theta)$.

Recordemos que $E[X] = 1/\theta$ y que $Var(X) = 1/\theta^2$. Además el primer momento muestra es $\bar{X}$ y despejando a $\theta$ se tiene que
$$\hat{\theta} = 1/\bar{X}$$

In [9]:
import numpy as np

# Datos proporcionados
datos = np.array([525, 719, 2880, 150, 30, 251, 45, 858, 15, 47, 90, 56, 68, 6, 189, 180, 60, 60, 294, 747])

#el tamaño de la muestra es 
n= len(datos)

# Calcular la media muestral
media_muestral = np.mean(datos)

# Calcular el estimador theta (1 / media)
theta_estimado = 1 / media_muestral

print(f"Media muestral: {media_muestral}")
print(f"Estimador theta: {theta_estimado:.6f}")

Media muestral: 363.5
Estimador theta: 0.002751


In [10]:
n

20

## <span style="color:#191970;">Interpretación</span>

Se tiene que $\hat{\theta} = \frac{1}{\bar{X}} \approx 0.00275$ por minuto.

Por lo tanto, estimamos que el tiempo promedio de retardo entre el inicio de los síntomas cardiovasculares y la llegada a la unidad de emergencia es de aproximadamente 363.5 minutos.

# <span style="color:#191970;">Ejemplo 5 </span>

Suponga que la vida útil de los focos de cierta marca, se modela mediante una distribución exponencial de parámetro $\theta$ (es desconocido). Probamos 5 focos y encontramos que tienen una vida útil de 2, 3, 1, 3 y 5 años respectivamente. Calcular el EMV (MLE) para $\theta$.

In [11]:
# Datos:
datos = np.array([2,3,1,3,5])
n = len(datos)
media = np.mean(datos)
# Calculamos el EMV para la distribución exponencial
theta = 1 / media

print(f"La media muestral (X barra) es: {media}")
print(f"El Estimador Máximo Verosímil (EMV) para theta es: {theta:.4f}")

La media muestral (X barra) es: 2.8
El Estimador Máximo Verosímil (EMV) para theta es: 0.3571


## <span style="color:#191970;">Interpretación</span>

Por lo tanto, la tasa estimada es de 0.3571 fallas por año y se  tiene que la vida media estimada de los focos es de 2,8 años

# <span style="color:#191970;">Ejercicio 6. Encontrar estimador con Sympy </span>


In [13]:

# Definimos las variables simbólicas
theta = sympy.symbols('theta', positive=True)
x = sympy.symbols('x', real=True)

# Definimos la función de verosimilitud para una sola observación (Exponencial)
f = theta * sympy.exp(-theta * x)

# Calculamos el producto de todos los elementos usando la muestra 'datos'
# subs(x,i) sustituye la variable simbólica x por cada valor concreto de la muestra
J = np.prod([f.subs(x, i) for i in datos])

# Calculamos el log de la verosimilitud
log_J = sympy.log(J)

# Calculamos la derivada respecto a theta
derivada = sympy.diff(log_J, theta)

# Resolvemos para hallar el EMV (igualamos a cero y despejamos theta)
solucion = sympy.solve(derivada, theta)

# Mostramos el resultado
print("Estimador Máximo Verosímil (EMV):")
solucion[0]

Estimador Máximo Verosímil (EMV):


5/14

## <span style="color:#191970;">Interpretación</span>

Se obtuvo un estimador exacto de 5/14 (equivalente a 0.3571).
Al realizar la optimización matemática simbólica, el sistema confirma de forma precisa que $\hat{\theta} = 5/14$ es el valor que maximiza la función de verosimilitud para nuestra muestra específica $\{2, 3, 1, 3, 5\}$. Esto valida que el cálculo numérico anterior es correcto y refleja el punto exacto donde la probabilidad de observar esos datos es máxima.

---
# **<span style="color:#191970;">Intervalos de Confianza</span>**

Un intervalo de confianza es un rango de valores que, con una cierta probabilidad (nivel de confianza), contiene al verdadero parámetro poblacional que queremos estimar.

* **Error Estándar (SE):** Mide cuánto varía un estimador.
    * Si conocemos $$\sigma: SE = \sigma / \sqrt{n}$$
    * Si NO conocemos $\sigma$, utilizamos $$s: SE = s / \sqrt{n}$$
* **Nota:** A mayor tamaño de muestra, el SE disminuye, haciendo el intervalo más preciso.

---

# <span style="color:#191970;">1. Intervalo para la media de una población normal</span>

Sea $X_1, \dots, X_n$ una muestra aleatoria con $X \sim N(\mu, \sigma^2)$.

*   $\bar{X} \sim \mathcal{N}(\mu, \sigma^2/n)$
*   $Z = \frac{\bar{X} - \mu}{\sigma/\sqrt{n}} \sim \mathcal{N}(0, 1)$

### <span style="color:#191970;">A) σ conocida (Distribución Z)</span>

Es el caso ideal donde se conoce la desviación estándar poblacional real ($\sigma$). Se utiliza la distribución Normal estándar ($Z$) porque no existe incertidumbre adicional al estimar la variabilidad.

Se aplica cuando el proceso es estable y su variabilidad histórica es un dato fijo conocido.

Cuando la varianza poblacional es conocida, el estadístico pivote $Z = \frac{\bar{X} - \mu}{\sigma/\sqrt{n}}$ sigue una distribución Normal estándar $N(0, 1)$. Esto permite definir el margen de error como el producto del valor crítico $z_{\alpha/2}$ y el error estándar.

$$IC = \bar{X} \pm z_{\alpha/2} \left( \frac{\sigma}{\sqrt{n}} \right)$$

donde

*   $z_{\alpha/2} \frac{\sigma}{\sqrt{n}}$ es el margen de error
*   $z_{\alpha/2}$ es el valor crítico
  
*Frase explicativa: Tenemos un $(1-\alpha)\%$ de confianza de que la media real se encuentra en este rango.*



### <span style="color:#191970;">Ejercicio 1.</span>

Para tratar de estimar la media de consumo por cliente en un restaurante, se reunieron datos de una muestra de tamaño 49 clientes durante tres semanas.

**a)** Supongamos que la desviación estándar poblacional es de \$2.5. ¿Cuál es el error estándar de la media?

**b)** Con un nivel de confianza del 95%, ¿cuál es el margen de error?

**c)** Si la media muestral es de \$22.60, ¿cuál es el intervalo de confianza del 95% para la media poblacional?

In [2]:
import numpy as np
import scipy.stats as st 

# Datos del problema
n = 49 # numero de clientes
sigma = 2.5
confianza = 0.95
media_muestral = 22.60

# a) Error estándar de la media (SE = sigma / sqrt(n))
error_est = sigma / np.sqrt(n)
print(f"a) El error estándar de la media es: {error_est:.4f}")

# b) Margen de error
# Usando st tal como lo hace tu profesora:
z = st.norm.ppf(1 - (1 - confianza) / 2)
margen_error = z * error_est

print(f"b) El valor crítico z es: {z:.4f}")
print(f"   El margen de error es: {margen_error:.4f}")

# c) Intervalo de confianza
limite_inferior = media_muestral - margen_error
limite_superior = media_muestral + margen_error
print(f"c) El intervalo de confianza del 95% es: ({limite_inferior:.2f}, {limite_superior:.2f})")

a) El error estándar de la media es: 0.3571
b) El valor crítico z es: 1.9600
   El margen de error es: 0.7000
c) El intervalo de confianza del 95% es: (21.90, 23.30)


#### <span style="color:#191970;">Interpretación</span>

* Error Estándar ($0.3571$): Indica que la media de nuestra muestra de 49 clientes tiende a variar en aproximadamente $0.3571$ unidades respecto a la verdadera media poblacional.
* Margen de Error ($0.7000$): Es la "tolerancia" de nuestra estimación; significa que, con un 95% de confianza, nuestra media muestral de $\$22.60$ podría desviarse hacia arriba o hacia abajo hasta un máximo de $\$0.70$ respecto al consumo real promedio de toda la población de clientes.
* Intervalo de Confianza ($21.98, 23.30$):Tenemos un 95% de confianza de que el consumo promedio real de todos los clientes del restaurante se encuentra entre $\$21.98$ y $\$23.30$.

Con un nivel de confianza del 95%, el gasto promedio real por cliente en el restaurante está contenido dentro de este intervalo.

### <span style="color:#191970;">Ejercicio 2</span>

Una empresa de logística quiere estimar el tiempo promedio de entrega de paquetes en la ciudad. Se sabe por registros históricos que la desviación estándar poblacional es de **15 minutos**. Se toma una muestra de **64 entregas** y se obtiene un promedio muestral de **45 minutos**.

Calcular el intervalo de confianza del 95% para el tiempo medio de entrega.

> 1. Identificar parámetros: $\sigma = 15$, $n = 64$, $\bar{X} = 45$.
> 2. Calcular el error estándar: $SE = \sigma / \sqrt{n}$.
> 3. Determinar el margen de error: $z_{\alpha/2} \times SE$ (donde $z \approx 1.96$ para 95% de confianza).
> 4. Construir el intervalo: $\bar{X} \pm \text{margen de error}$.

In [5]:
# Datos del problema
n = 64
sigma = 15
media_muestral = 45
confianza = 0.95

# Cálculos
error_est = sigma / np.sqrt(n)
z = st.norm.ppf(1 - (1 - confianza) / 2)
margen_error = z * error_est

limite_inferior = media_muestral - margen_error
limite_superior = media_muestral + margen_error

print(f"El error estándar de la media es: {error_est:.4f}")
print(f"El margen de error es: {margen_error:.4f}")
print(f"El intervalo de confianza del 95% es: ({limite_inferior:.2f}, {limite_superior:.2f})")

El error estándar de la media es: 1.8750
El margen de error es: 3.6749
El intervalo de confianza del 95% es: (41.33, 48.67)


#### <span style="color:#191970;">Interpretación</span>
Con una muestra de 64 entregas, obtuvimos un margen de error de 3.6750 minutos, resultando en un intervalo de confianza de (41.33, 48.67) minutos.

El intervalo nos permite asegurar que, basándonos en nuestra muestra y considerando la variabilidad histórica conocida, el tiempo promedio real de entrega de todos los paquetes de la ciudad se encuentra entre 41.33 y 48.67 minutos.

Con un nivel de confianza del 95%, podemos afirmar que el tiempo medio poblacional de entrega está contenido dentro de este rango de tiempo.

### <span style="color:#191970;">B) σ desconocida (Distribución t de Student)</span>

Es el escenario más frecuente en la práctica estadística, donde el valor de la desviación estándar poblacional ($\sigma$) es desconocido y debe ser estimado mediante la desviación estándar de la muestra ($s$).

Se utiliza cuando trabajamos con muestras nuevas o estudios exploratorios donde no se dispone de información histórica sobre la variabilidad de la población.

Al no conocer $\sigma$, usamos la desviación muestral ($s$). La distribución $t$ compensa la mayor incertidumbre usando colas más anchas.
Utilizamos la desviación muestral $s$ y el valor crítico $t_{\alpha/2, n-1}$:
$$IC = \bar{X} \pm t_{\alpha/2, n-1} \left( \frac{s}{\sqrt{n}} \right)$$
> **Interpretación:** Este intervalo es más conservador, compensando la incertidumbre de no conocer la varianza real con colas más pesadas en la distribución $t$.

> Frase explicativa: Con un nivel de confianza del $(1-\alpha)\%$, el valor real de la media poblacional reside dentro de este intervalo calculado.*



### <span style="color:#191970;">Ejercicio 3.</span>

Supongamos que tenemos los siguientes datos de una muestra aleatoria de 10 observaciones:

$$45, 55, 67, 68, 45, 79, 98, 87, 84, 82$$

Calcular un intervalo de confianza para la media

In [3]:
# Datos
datos = [45,55,67,68,45,79,98,87,84,82] # n<= 30
n = len(datos)
confianza = 0.95
gl = n-1

media_muestral = np.mean(datos)
error_est = st.sem(datos) # eror estandar muestral

# Vamos a utilizar la función 
# scipy.stats.t.interval(confidence, df, loc, scale)
# Cofidence =cofianza
# df =degre freedom (log)
# loc= media_muestral
# scale=error_est

intervalo = st.t.interval(confianza, gl, media_muestral,error_est)
print(f"Intervalo de confianza completo del {confianza* 100}%:( {intervalo[0]:.2f},{intervalo[1]:.2f})")

Intervalo de confianza completo del 95.0%:( 58.00,84.00)


#### <span style="color:#191970;">Interpretación</span>
* Para la muestra de 10 datos, el intervalo de confianza del 95% es (58.00, 84.00).
* Aunque no conocemos la desviación estándar poblacional, hemos estimado que el promedio real de la población se encuentra dentro de este rango con un 95% de confianza. La amplitud del intervalo refleja la variabilidad presente en tus 10 observaciones.
  
* **Frase clave:** Con un nivel de confianza del 95%, el valor verdadero de la media poblacional reside dentro del intervalo calculado de 58.00 a 84.00.

### <span style="color:#191970;">Ejercicio 4: Tiempo de espera </span>
Este ejercicio te servirá para aplicar lo aprendido cuando no conoces la varianza poblacional.

Problema: Un banco desea estimar el tiempo promedio de espera de sus clientes. Se toma una muestra aleatoria de 12 clientes y se obtienen los siguientes tiempos en minutos: 
$$[10, 12, 15, 14, 18, 11, 13, 19, 10, 16, 12, 14]$$

Calcula el intervalo de confianza del 95% para la media.

In [6]:
# Datos del problema
tiempos = [10, 12, 15, 14, 18, 11, 13, 19, 10, 16, 12, 14]
n = len(tiempos)
confianza = 0.95
gl = n - 1

# Cálculos
media_muestral = np.mean(tiempos)
error_est = st.sem(tiempos) # Calcula el error estándar de la muestra

# Cálculo del intervalo con t de Student
intervalo_t = st.t.interval(confianza, gl, loc=media_muestral, scale=error_est)

print(f"El promedio muestral es: {media_muestral:.2f}")
print(f"Intervalo de confianza del 95% (t-Student): ({intervalo_t[0]:.2f}, {intervalo_t[1]:.2f})")

El promedio muestral es: 13.67
Intervalo de confianza del 95% (t-Student): (11.80, 15.53)


#### <span style="color:#191970;">Interpretación</span>

 
Para una muestra de 12 clientes, se obtuvo una media de 13.67 minutos, con un intervalo de confianza del 95% de (11.80, 15.53) minutos.

Dado que desconocemos la variabilidad total de todos los clientes del banco, utilizamos la distribución $t$ de Student, lo que nos brinda una estimación más prudente y ajustada a la incertidumbre de nuestra muestra pequeña.

* **Frase clave:** Con un nivel de confianza del 95%, podemos afirmar que el tiempo promedio real de espera de todos los clientes del banco se encuentra contenido dentro de este intervalo de 11.80 a 15.53 minutos.

---

## <span style="color:#191970;">2. Intervalo para la Varianza ($\sigma^2$)</span>

Es el método para estimar la dispersión o consistencia de una población. A diferencia de la media, la varianza mide qué tanto se alejan los datos individuales del promedio.

Es fundamental en el Control de Calidad, donde no solo importa que el promedio de un producto sea correcto, sino que todos los productos sean uniformes (baja variabilidad).

Cuando $\mu$ es desconocida, el intervalo de confianza para la varianza poblacional $\sigma^2$ se basa en la distribución **Chi-cuadrada ($\chi^2$)** con $n-1$ grados de libertad:

$$IC = \left( \frac{(n-1)s^2}{\chi^2_{\alpha/2}}, \frac{(n-1)s^2}{\chi^2_{1-\alpha/2}} \right)$$



---

## <span style="color:#191970;">3. Intervalo para una Proporción ($p$)</span>

Es el método para estimar el porcentaje o proporción de elementos en una población que poseen una característica específica (ej. éxito, preferencia, defecto).

Se utiliza en encuestas de opinión, estudios de mercado y control de procesos por atributos (pasa/no pasa).

Para datos categóricos (muestras grandes):
$$IC = \hat{p} \pm z_{\alpha/2} \sqrt{\frac{\hat{p}(1-\hat{p})}{n}}$$
*Nota: Existe un $(1-\alpha)\%$ de confianza de que la proporción verdadera de la población está contenida en este intervalo.*

---
Tabla Resumen de Inferencia

| Parámetro | Estadística | Distribución |
| :--- | :--- | :--- |
| Media ($\sigma$ conocida) | $\bar{X}$ | Normal ($Z$) |
| Media ($\sigma$ desconocida) | $\bar{X}$ | $t$ de Student |
| Varianza ($\mu$ desconocida) | $s^2$ | Chi-cuadrada ($\chi^2$) |
| Proporción | $\hat{p}$ | Normal ($Z$) |

### <span style="color:#191970;">Ejercicio 5.</span>
Los artículos de cerámica utilizados sobre velas eléctricas se rompen con diferentes presiones. Supongamos que los datos provienen de una distribución normal. La resistencia a la ruptura fue medida en una muestra de 100 artículos y el promedio fue de 1750 con una desviación estándar de 315.8.

**a)** Estimar con un nivel de confianza del 90% a la media poblacional de la presión de la ruptura.

**b)** Estimar con un nivel de confianza del 90% a la varianza poblacional.

In [4]:
# Datos
n = 100
confianza = 0.90
media_muestral = 1750
desv_est = 315.8
error_est = desv_est / np.sqrt(n)

# --- Variables que faltaban definir en tus datos ---
alpha = 1 - confianza
gl = n - 1
s2 = desv_est ** 2  

# Forma 1. Calculando los cuatiles de la chi cuadrada
chi_inf = st.chi2.ppf(alpha/2, gl)
chi_sup = st.chi2.ppf(1-alpha/2, gl)
int_var = ((gl*s2) / chi_sup, (gl*s2) / chi_inf)
print(f'A) El intervalo de confianza del {confianza*100}%: ({int_var[0]:.2f},{int_var[1]:.2f})')


# Forma 2.
chi2_inf, chi2_sup = st.chi2.interval(confianza, gl)
int2_var = ((gl*s2) / chi2_sup, (gl*s2) / chi2_inf)
print(f'B) El intervalo de confianza del {confianza*100}%: ({int2_var[0]:.2f},{int2_var[1]:.2f})')

A) El intervalo de confianza del 90.0%: (80123.49,128146.72)
B) El intervalo de confianza del 90.0%: (80123.49,128146.72)


#### <span style="color:#191970;">Interpretación</span>

 
* **a) Media poblacional:** Para una muestra de 100 artículos con un promedio de 1750, el intervalo de confianza del 90% (calculado mediante la distribución normal, ya que $n \geq 30$) estima dónde se encuentra la resistencia media real a la ruptura de los artículos de cerámica.
* **b) Varianza poblacional ($\sigma^2$):** Para la misma muestra, obtuvimos un intervalo de confianza del 90% de (80123.49, 128146.72).
*  Este intervalo nos permite conocer la dispersión o consistencia del proceso de fabricación. Al ser una medida de variabilidad, nos indica que con un 90% de confianza, la varianza real de la presión de ruptura de la cerámica está contenida en este rango.
* **Frase clave:** Existe un 90% de confianza de que la variabilidad real (varianza) de la población se encuentra entre 80123.49 y 128146.72 unidades al cuadrado.

### <span style="color:#191970;">Ejercicio 6.</span>
Una empresa de marketing quiere estimar qué proporción de usuarios prefiere su nueva aplicación móvil. Se realizó una encuesta a 200 usuarios, de los cuales 120 indicaron que prefieren la nueva app sobre la anterior.
Calcular el intervalo de confianza del 95% para la proporción poblacional ($p$).

In [7]:
# Datos del problema
n = 200
x = 120 # numero de exitos
p_hat = x / n # Proporción muestral
confianza = 0.95

# Cálculo del margen de error (usando la distribución normal Z)
z = st.norm.ppf(1 - (1 - confianza) / 2)
error_est = np.sqrt((p_hat * (1 - p_hat)) / n)
margen_error = z * error_est

# Cálculo del intervalo
limite_inferior = p_hat - margen_error
limite_superior = p_hat + margen_error

print(f"La proporción muestral es: {p_hat:.2f}")
print(f"Intervalo de confianza del 95%: ({limite_inferior:.4f}, {limite_superior:.4f})")

La proporción muestral es: 0.60
Intervalo de confianza del 95%: (0.5321, 0.6679)


#### <span style="color:#191970;">Interpretación</span>

Con una muestra de 200 usuarios y una proporción muestral de 0.60, el intervalo de confianza del 95% es (0.5325, 0.6675).

Este intervalo estima la proporción real de todos los usuarios que prefieren la nueva aplicación. Dado que el intervalo está por encima de 0.50, podemos concluir con alta confianza que más de la mitad de los usuarios prefieren la nueva versión.
* **Frase clave:** Existe un 95% de confianza de que la proporción verdadera de usuarios que prefieren la nueva aplicación se encuentra entre el 53.25% y el 66.75%.

----
# **<span style="color:#191970;">Conclusión</span>**

La inferencia estadística constituye una herramienta esencial para obtener información confiable sobre una población a partir del análisis de una muestra. Su importancia radica en que permite realizar estimaciones y tomar decisiones fundamentadas aun cuando no es posible estudiar a todos los elementos de una población.

En este proceso, los estimadores desempeñan un papel fundamental, ya que proporcionan una aproximación de los parámetros poblacionales desconocidos. No obstante, para garantizar que dichas estimaciones sean confiables, es indispensable evaluar sus propiedades estadísticas, como el insesgamiento, que evita errores sistemáticos; la eficiencia, que busca minimizar la variabilidad de las estimaciones; y la consistencia, que asegura que el estimador se aproxime al valor verdadero del parámetro conforme aumenta el tamaño de la muestra.

Asimismo, los intervalos de confianza complementan la estimación puntual al expresar el grado de incertidumbre inherente al proceso de muestreo. Su correcta interpretación consiste en entender que representan un rango de valores plausibles para el parámetro poblacional con un nivel de confianza previamente establecido, lo que permite valorar la precisión de las estimaciones y respaldar la toma de decisiones con mayor fundamento estadístico.

En conclusión, comprender el papel de los estimadores, evaluar las propiedades que determinan su calidad e interpretar adecuadamente los intervalos de confianza constituye la base de una inferencia estadística sólida. Estos conceptos permiten transformar la información muestral en conocimiento útil, objetivo y científicamente sustentado, fortaleciendo la validez de los análisis y la calidad de las decisiones en ámbitos académicos, científicos y profesionales.

---

# **<span style="color:#191970;"> Bibliográfias</span>**

* Rincón, L. (2014). Introducción a la probabilidad (1ª ed.). Universidad Nacional Autónoma de México, Facultad de Ciencias.
* Mood, A. M., Graybill, F. A., y Boes, D. C. (1974). Introduction to the theory of statistics. McGraw-Hill.
* Ross, S. M. (2010). A first course in probability (8ª ed.). Pearson Prentice Hall.
* Wackerly, D. D., Mendenhall, W., y Scheaffer, R. L. (2010). Estadística matemática con aplicaciones (7ª ed.). Cengage Learning
* Rincón, L. (2019). Una introducción a la estadística inferencial. Universidad Nacional Autónoma de México, Facultad de Ciencias.
---